src/rust/engine/src/context.rs

```diff
@@ -99,6 +99,171 @@ pub struct ExecutionStrategyOptions {
 }
 
 impl Core {
+  fn make_store(
+    executor: &Executor,
+    local_store_dir: &Path,
+    enable_remote: bool,
+    remoting_opts: &RemotingOptions,
+    remote_store_servers: &[String],
+    root_ca_certs: &Option<Vec<u8>>,
+    oauth_bearer_token: &Option<String>,
+  ) -> Result<Store, String> {
+    if enable_remote {
+      Store::with_remote(
+        executor.clone(),
+        local_store_dir,
+        remote_store_servers.to_vec(),
+        remoting_opts.instance_name.clone(),
+        root_ca_certs.clone(),
+        oauth_bearer_token.clone(),
+        remoting_opts.store_thread_count,
+        remoting_opts.store_chunk_bytes,
+        remoting_opts.store_chunk_upload_timeout,
+        // TODO: Take a parameter
+        store::BackoffConfig::new(Duration::from_millis(10), 1.0, Duration::from_millis(10))
+          .unwrap(),
+        remoting_opts.store_rpc_retries,
+        remoting_opts.store_connection_limit,
+      )
+    } else {
+      Store::local_only(executor.clone(), local_store_dir.to_path_buf())
+    }
+  }
+
+  fn make_local_execution_runner(
+    store: &Store,
+    executor: &Executor,
+    local_execution_root_dir: &Path,
+    named_caches_dir: &Path,
+    process_execution_metadata: &ProcessMetadata,
+    exec_strategy_opts: &ExecutionStrategyOptions,
+  ) -> Result<Box<dyn CommandRunner>, String> {
```

Comment:
does this need to return a `Result`? I don't see a way this can fail

Reply:
I wanted to be consistent with the other refactored methods. They all return `Result<_, String>`. You are correct though, as it is currently, it cannot fail.

Reply:
Let's not return a `Result` here - can always change the signature if we need to in the future :)

Reply:
Done. Now just `Box<dyn CommandRunner>`.